core/src/impl/Cabana_Matrix2d.hpp

```diff
@@ -0,0 +1,210 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+/*!
+  \file impl/Cabana_Matrix2d.hpp
+  \brief
+  Linear algebra helpers. A lot of them are not implemented for generic
+  matricies and we probably shouldn't do that either, but call to some
+  appropriate linear algebra package. Special cases for small matricies are
+  implemented below.
+*/
+#ifndef CABANA_MATRIX2D_IMPL_HPP
+#define CABANA_MATRIX2D_IMPL_HPP
+
+#include <cmath>
+
+namespace Cabana {
+
+namespace Impl {
+
+template<typename FloatType, int dim>
+class Matrix2d {
```

Comment:
We have a relatively complete set of linear algebra in a separate repo that we've considered moving (mostly just FYI): https://github.com/picassodev/picasso/blob/master/unit_test/tstBatchedLinearAlgebra.hpp

Reply:
I'm totally happy with removing my code once that lands.